src/Ipac_Table_Writer/internals.cxx

```diff
@@ -108,94 +192,257 @@ void tablator::Ipac_Table_Writer::write_selected_records(
+    if (cols_begin != included_column_ids.end() && *cols_begin == 0) {
+        ++cols_begin;
+    }
+
+    os << "|";
+    os << std::right;
+
+    // Write column names
+    std::vector<size_t>::const_iterator cols_iter = cols_begin;
+    for (/* */; cols_iter != included_column_ids.end(); ++cols_iter) {
+        auto col_id = *cols_iter;
+        size_t effective_array_size = get_effective_array_size(
+                datatypes_for_writing[col_id], columns[col_id].array_size);
+        total_record_width += write_column_name(
+                table, os, col_id, ipac_column_widths[col_id], effective_array_size);
+    }
+    os << "\n|";
+
+    // Write column types
+    cols_iter = cols_begin;
+    for (/* */; cols_iter != included_column_ids.end(); ++cols_iter) {
+        auto col_id = *cols_iter;
+        write_column_type(table, os, datatypes_for_writing, col_id,
+                          ipac_column_widths[col_id]);
+    }
+
+    os << "\n|";
+
+    // Write column units
+    cols_iter = cols_begin;
+    for (/* */; cols_iter != included_column_ids.end(); ++cols_iter) {
+        auto col_id = *cols_iter;
+        size_t effective_array_size = get_effective_array_size(
+                datatypes_for_writing[col_id], columns[col_id].array_size);
+        write_column_unit(table, os, col_id, ipac_column_widths[col_id],
+                          effective_array_size);
     }
     os << "\n|";
 
     // Write column null property
-    for (size_t i = 1; i < columns.size(); ++i) {
-        auto column = columns[i];
+    cols_iter = cols_begin;
+    if (cols_begin != included_column_ids.end() && *cols_begin == 0) {
+        ++cols_begin;
+    }
+    for (/* */; cols_iter != included_column_ids.end(); ++cols_iter) {
+        auto col_id = *cols_iter;
+        size_t effective_array_size = get_effective_array_size(
+                datatypes_for_writing[col_id], columns[col_id].array_size);
+        write_column_null(table, os, col_id, ipac_column_widths[col_id],
+                          effective_array_size);
+    }
+    os << "\n";
+}
+
+
+/**********************************************************/
+size_t tablator::Ipac_Table_Writer::write_column_name(const Table& table,
+                                                      std::ostream& os, size_t col_id,
+                                                      size_t col_width,
+                                                      size_t effective_array_size) {
+    auto& column = table.columns[col_id];
+    size_t total_width;
+    if (effective_array_size == 1) {
+        total_width = (col_width + 1);
+        os << std::setw(col_width) << column.name << "|";
+    } else {
+        total_width = (col_width + 1) * column.array_size;
+        for (size_t element = 0; element < column.array_size; ++element) {
+            os << std::setw(col_width) << (column.name + "_" + std::to_string(element))
+               << "|";
```

Comment:
Just out of curiosity - does the column width computation take the maximum length of the _xxx suffix being added here into account?

Reply:
Yes, get_column_widths() does that.  Glad you asked, because I was assuming that that function was just returning a class member, but in fact it was loading as well as returning the vector.